golang/cosmos/x/swingset/config.go

```diff
@@ -0,0 +1,45 @@
+package swingset
+
+import (
+	"fmt"
+	"github.com/spf13/viper"
+	"path/filepath"
+)
+
+const (
+	ConfigPrefix = "swingset"
+	FlagSlogfile = "swingset.slogfile"
+)
+
+// DefaultConfigTemplate defines a default TOML configuration section for the SwingSet VM.
+// See https://github.com/cosmos/cosmos-sdk/issues/20097 for auto-synchronization ideas.
+const DefaultConfigTemplate = `
+
+	[swingset]
+	# slogfile is the absolute path at which a SwingSet log "slog" file should be written.
+	slogfile = ""
```

Comment:
Nit: an absolute file path is kinda weird, wondering if we could somehow resolve it relative to the home dir

Reply:
I'm not sure that's a great referent. And we can always relax this later.

Reply:
When I mentioned home dir, it's the cosmos home dir, aka `~/.agoric` by default. Why wouldn't that be a good referent?

Reply:
> When I mentioned home dir, it's the cosmos home dir, aka ~/.agoric by default. Why wouldn't that be a good referent?

For starters, because I think we have multiple poorly-explained concepts of "home dir"—distinct from cosmos-sdk `FlagHome` [--home] but used to set its default value is our golang/cosmos/app/app.go `DefaultNodeHome` "_default home directories for the application daemon_", which is initialized to ~/.agoric but set to ~/.ag-chain-cosmos by `RunAgCosmosDaemon` for the goroutine that runs inside Node.js. Our app initialization uses `FlagHome` (including to set the default SwingStoreExportDir to ./config/swing-store underneath that), but most other code uses `DefaultNodeHome` (_including the syscall.Exec that replaces agd with Node.js ag-chain-cosmos, which I just realized is a bug because `agd --home=/path/to/dir start` → `ag-chain-cosmos --home ~/.agoric --home /path/to/dir start` ✔ but `AGD_HOME=/path/to/dir agd start` → `ag-chain-cosmos --home ~/.agoric start` ✘_).

But beyond that, because interpretation of relative paths in config files is just inherently confusing unless very well explained (and generally best when relative to the file itself). But I will add support for interpretation relative to current directory for values specified dynamically via e.g. environment variable.

Reply:
Yeah the handling of the `home` env/flag could be improved, but besides that, it really seems that every relative path in the config is resolved from that folder. For example, take the `add_book_file`, `node_key_file`, `genesis_file` etc. from `config.toml`. They are all relative to the cosmos home dir. In `app.toml`, the `streamers.file.write_dir` is also resolved relative to the home dir: https://github.com/agoric-labs/cosmos-sdk/blob/2c7e4030ee73d926bb979460fcab1cd769d93bd9/store/streaming/constructor.go#L101-L104

Reply:
OK, that's persuasive. Done.